bindep.txt

```diff
@@ -0,0 +1 @@
+kubernetes-client [platform:fedora]
```

Comment:
`kubectl` is not a requirement of this collection; you would only need it if you had tasks that used `command` or `shell` and called out to it directly.

Reply:
@geerlingguy It is used for the connection plugin right?

Reply:
https://github.com/ansible-collections/community.kubernetes/blob/main/plugins/connection/kubectl.py#L36

Reply:
D'oh, you're right: https://github.com/ansible-collections/community.kubernetes/blob/main/plugins/connection/kubectl.py#L36

Reply:
It's here

https://github.com/ansible-collections/community.kubernetes/blob/e9d8bf703bcb9bced1ae327d99be99e907d7d83a/plugins/connection/kubectl.py#L35-L36

There's a big issue here between requirements _of a module_, _of a plugin_, and _of a collection_.

I have not formalized this yet, but the argument I want to make is that plugins which have to run on the _control_ machine should be a minimum to say that a collection can build a "full" execution environment. Modules may expect a dependency to be present on the remote machine, but not the control machine.

A connection plugin that requires `kubectl` is clearly on the side of the control machine. So would action plugins, inventory plugins, etc. I want to formalize guidelines to lays this out and would love to get you involved.